extensions/authentication-jwt/README.md

```diff
@@ -0,0 +1,209 @@
+
+<details>
+<summary><strong>Check The Code</strong></summary>
+<p>
+
+```ts
+export class TestApplication extends BootMixin(
+  ServiceMixin(RepositoryMixin(RestApplication)),
+) {
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+
+    // Set up the custom sequence
+    this.sequence(MySequence);
+
+    // Set up default home page
+    this.static('/', path.join(__dirname, '../public'));
+
+    // - enable jwt auth -
+    // Add security spec (Future work: refactor it to an enhancer)
+    this.addSecuritySpec();
+    // Mount authentication system
+    this.component(AuthenticationComponent);
+    // Mount jwt component
+    this.component(JWTAuthenticationComponent);
+    // Bind datasource
+    this.dataSource(DbDataSource, UserServiceBindings.DATASOURCE_NAME);
+
+    this.component(RestExplorerComponent);
+    this.projectRoot = __dirname;
+    // Customize @loopback/boot Booter Conventions here
+    this.bootOptions = {};
+  }
+
+  // - enable jwt auth -
+  // Currently there is an extra function to
+  // merge the security spec into the application.
+  // This will be improved with a coming enhancer.
+  // See section [Future Work](#future-work)
+  addSecuritySpec(): void {
+    this.api({
+      openapi: '3.0.0',
+      info: {
+        title: 'test application',
+        version: '1.0.0',
+      },
+      paths: {},
+      components: {securitySchemes: SECURITY_SCHEME_SPEC},
+      security: [
+        {
+          // secure all endpoints with 'jwt'
+          jwt: [],
+        },
+      ],
+      servers: [{url: '/'}],
+    });
+  }
+}
+```
+
+</p>
+</details>
+
+_All the jwt authentication related code are marked with comment "- enable jwt
+auth -", you can search for it to find all the related code you need to enable
+the entire jwt authentication in a LoopBack 4 application._
+
+## Add Endpoint in Controller
+
+After mounting the component, you can call token and user services to perform
+login then decorate endpoints with `@authentication('jwt')` to inject the logged
+in user's profile.
+
+This module contains an example LoopBack 4 application in folder `fixtures`. It
+has a controller with endpoints `/login` and `/whoAmI` for demo's purpose.
+
+The code snippet for login function:
+
+<details>
+<summary><strong>Check The Code</strong></summary>
+<p>
+
+```ts
+async login(
+    @requestBody(CredentialsRequestBody) credentials: Credentials,
+  ): Promise<{token: string}> {
+    // ensure the user exists, and the password is correct
+    const user = await this.userService.verifyCredentials(credentials);
+
+    // convert a User object into a UserProfile object (reduced set of properties)
+    const userProfile = this.userService.convertToUserProfile(user);
+
+    // create a JSON Web Token based on the user profile
+    const token = await this.jwtService.generateToken(userProfile);
+
+    return {token};
+  }
+```
+
+</p>
+</details>
+
+The code snippet for whoAmI function:
+
+<details>
```

Comment:
Sometimes it's better to show the code by default, especially if its not a lot. I'd suggest to remove `<details>`.